src/test/java/seedu/address/storage/JsonAdaptedLocalCourseTest.java

```diff
@@ -26,31 +27,52 @@ public void toModelType_validLocalCourseDetails_returnsLocalCourse() throws Exce
     @Test
     public void toModelType_invalidLocalCode_throwsIllegalValueException() {
         JsonAdaptedLocalCourse localCourse =
-                new JsonAdaptedLocalCourse(INVALID_LOCAL_COURSE_CODE, TYPICAL_LOCAL_COURSE_NAME);
+                new JsonAdaptedLocalCourse(
+                    INVALID_LOCAL_COURSE_CODE,
+                    TYPICAL_LOCAL_COURSE_NAME,
+                    TYPICAL_LOCAL_COURSE_UNIT);
         String expectedMessage = LocalCode.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, localCourse::toModelType);
     }
 
     @Test
     public void toModelType_nullLocalCode_throwsIllegalValueException() {
-        JsonAdaptedLocalCourse localCourse = new JsonAdaptedLocalCourse(null, TYPICAL_LOCAL_COURSE_NAME);
+        JsonAdaptedLocalCourse localCourse = new JsonAdaptedLocalCourse(
+            null,
+            TYPICAL_LOCAL_COURSE_NAME,
+            TYPICAL_LOCAL_COURSE_UNIT);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, LocalCode.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, localCourse::toModelType);
     }
 
     @Test
     public void toModelType_invalidLocalName_throwsIllegalValueException() {
         JsonAdaptedLocalCourse localCourse =
-                new JsonAdaptedLocalCourse(TYPICAL_LOCAL_COURSE_CODE, INVALID_LOCAL_COURSE_NAME);
+                new JsonAdaptedLocalCourse(
+                    TYPICAL_LOCAL_COURSE_CODE,
+                    INVALID_LOCAL_COURSE_NAME,
+                    TYPICAL_LOCAL_COURSE_UNIT);
         String expectedMessage = LocalName.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, localCourse::toModelType);
     }
 
     @Test
     public void toModelType_nullLocalName_throwsIllegalValueException() {
-        JsonAdaptedLocalCourse localCourse = new JsonAdaptedLocalCourse(TYPICAL_LOCAL_COURSE_CODE, null);
+        JsonAdaptedLocalCourse localCourse = new JsonAdaptedLocalCourse(
+            TYPICAL_LOCAL_COURSE_CODE,
+            null,
+            TYPICAL_LOCAL_COURSE_UNIT);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, LocalName.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, localCourse::toModelType);
     }
 
+    @Test
+    public void toModelType_negativeLocalUnit_throwsIllegalValueException() {
+        JsonAdaptedLocalCourse localCourse = new JsonAdaptedLocalCourse(
+            TYPICAL_LOCAL_COURSE_CODE,
+            null,
+            TYPICAL_LOCAL_COURSE_UNIT);
+        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, LocalName.class.getSimpleName());
+        assertThrows(IllegalValueException.class, expectedMessage, localCourse::toModelType);
+    }
 }
```

Comment:
Great testing on the negative LocalUnit! Maybe can add tests for null value and non-string/number values also?